packages/devtools_app/lib/src/screens/network/network_screen.dart

```diff
@@ -137,6 +138,11 @@ class _NetworkScreenBodyState extends State<NetworkScreenBody>
 
   @override
   Widget build(BuildContext context) {
+    addAutoDisposeListener(serviceManager.isolateManager.mainIsolate, () {
```

Comment:
we should never be adding listeners or doing business logic like this in `build`. This would add a listener each time the widget is rebuilt, which is not what we want.

we should do this in didChangeDependencies after we make the initial call to controller.startRecording.

Reply:
After discussing this with Kenzie I've added this to the `didChangeDependencies` as well as added a call to `cancelListeners` to avoid adding duplicates :D